### ISSUE TITLE
Leaks in every filter when used via Cocoapods.(Workaround/Fix)

### ISSUE THREAD

**Issue by atishay** [BradLarson/GPUImage#1748]
Use OS_OBJECT_USE_OBJC instead of __IPHONE_OS_VERSION_MIN_REQUIRED < __IPHONE_6_0 for ARC check for GCD or drop iOS 5 altogether.

**TL; DR**
Coming of a 5 hour investigation. It is a weird combination of Cocoapods, GPUImage and ARC that causes this leak.
- GPUImage defines that it supports iOS 5.0 with ARC in its podspec. 
- iOS 5.0 does not have ARC for GCD(GrandCentral Dispatch). 
- GPUImage takes care of this by having `__IPHONE_OS_VERSION_MIN_REQUIRED < __IPHONE_6_0` checks. It does not use `OS_OBJECT_USE_OBJC` check which is the correct way to see if ARC for GCD is supported. 
- Although the target project that cocapods has is iOS7.1, but as GPUImage specifies the minimum as 5.0, cocoapods prepares GPUImage compile options based on 5.0. That causes all GCD objects to leak.

As a workaround, I have created a local podspec with ios minimum specified as iOS 7. This fixes the bug for me. I can create a pull request with the entire set of find replace throughout the code, but it seems like a waste of time as just changing a number in the podspec fixes this issue. (The shouldn't be a huge number of developers needing GPUImage against iOS5 SDK)
